### PR TITLE
Add methods of the checkout recurring

### PR DESCRIPTION
**Pull request by aait**

### CODE REVIEW

File: src/SwedbankPay/Core/Adapter/WC_Adapter.php [hunk 1]
```diff
@@ -549,7 +549,14 @@ public function savePaymentToken(
         $expiry_date = explode('/', $expiry_date);
 
         // Create Payment Token
-        $token = new WC_Payment_Token_Swedbank_Pay();
+        if (!property_exists($this->gateway, 'payment_token_class') ||
+            empty($this->gateway->payment_token_class)
+        ) {
+            throw new \Exception(__('Payment Token class is undefined.', 'swedbank-pay-woocommerce-payments'));
```
Comment (asbjornu) [inline src/SwedbankPay/Core/Adapter/WC_Adapter.php:555]: Will this now require that WooCommerce Subscriptions is installed, or under which circumstances is `$this->gateway->payment_token_class` set?

File: src/SwedbankPay/Core/Library/Methods/Checkout.php [hunk 2]
```diff
@@ -85,6 +86,55 @@ public function initiatePaymentOrderPurchase($orderId, $consumerProfileRef = nul
...
+    /**
+     * Initiate Payment Order Recurrent Payment
+     *
+     * @param mixed $orderId
+     * @param string $recurrenceToken
+     *
+     * @return Response
+     * @throws \Exception
+     */
+    public function initiatePaymentOrderRecur($orderId, $recurrenceToken)
```
Comment (asbjornu) [inline src/SwedbankPay/Core/Library/Methods/Checkout.php:98]: `initiatePaymentOrderRecur` feels like something that should be inside the SDK, not in Core.

**Review by asbjornu — CHANGES_REQUESTED** [review]

File: src/SwedbankPay/Core/Library/Methods/Checkout.php (hunk 2 above)
Comment (aait) [reply src/SwedbankPay/Core/Library/Methods/Checkout.php:98]: We wrap some helpful method and proxy it to the SDK. Don' t worry, we will do it in the SDK also

File: src/SwedbankPay/Core/Adapter/WC_Adapter.php (hunk 1 above)
Comment (aait) [reply src/SwedbankPay/Core/Adapter/WC_Adapter.php:555]: @asbjornu No, It won't reqture WC Subscription. That's like Vault of Magento 2. WooCommerce use it for handling of recurring tokens

**Review by asbjornu — APPROVED** [review]

**Comment by asbjornu** [comment]
Why are the tests failing?

**Comment by aait** [comment]
> Why are the tests failing?

I don't know. I was surpriced too. It seems there are some troubles on payex side

**Pull request merged**